modules/essentials/ecr.tf

```
@@ -36,3 +36,10 @@ resource "aws_iam_role_policy_attachment" "worker_ecr_pullthrough" {
   role       = var.worker_iam_role_name
   policy_arn = aws_iam_policy.ecr_cache[0].arn
 }
+
+resource "aws_iam_role_policy_attachment" "worker_ecr_existing_pullthrough" {
+  count = !var.configure_ecr_pull_through && var.attach_existing_ecr_policy ? 1 : 0
```

Why not only attach when `var.existing_ecr_policy_arn` is not empty.

```suggestion
  count = !var.configure_ecr_pull_through && var.existing_ecr_policy_arn != "" && var.existing_ecr_policy_arn != null ? 1 : 0
```

Can remove one variable.